### PR TITLE
Add documentation for doc flags

### PR DESCRIPTION
**Pull request by samueleaton**
Pending Crystal PR **[3519](https://github.com/crystal-lang/crystal/pull/3519)**

screen shot 2016-11-10 at 8 18 51 am

### CODE REVIEW

File: conventions/documenting_code.md [hunk 1]
```diff
@@ -113,6 +113,41 @@ class Unicorn
...
+- BUG
+- DEPRECATED
+- FIXME
+- NOTE
+- OPTIMIZE
+- TODO
+
+Flag keywords must be the first word in their respective line and must be in all caps.
+
+Although a trailing colon can help with readability in the code, it will be removed in the generated doc. A trailing description of the flag is optional (but helpful).
```
Comment (RX14) [inline conventions/documenting_code.md]: Personally I would merge these two paragraphs as:
> Flag keywords must be the first word in their respective line and must be in all caps. An optional trailing colon is preferred for readability.

then let the example do the rest of the talking.

Comment (samueleaton) [inline conventions/documenting_code.md]: Less is more. I like it.

**Comment by Sija** [comment]
GTG since https://github.com/crystal-lang/crystal/pull/3519 is merged.

**Comment by asterite** [comment]
Thank you!

**Pull request merged**